neo4j/assets/configuration/spec.yaml

```
@@ -1,32 +1,82 @@
 name: Neo4j
+
 files:
 - name: neo4j.yaml
+  
   options:
   - template: init_config
-    options:
-    - template: init_config/http
-    - template: init_config/default
+        
   - template: instances
     options:
-    - name: neo4j_url
+    - name: host
       required: true
-      description: |
-        Neo4J url to connect to.
+      description: Neo4j host to connect to
       value:
-        type: string
-        example: localhost
+              type: string
+              example: localhost
+              
     - name: port
-      description: |
-        Port to connect to when reaching `neo4j_url`
+      required: false
+      description: Port which prometheus is listening on
+      value:
+              type: integer
+              example: 2004
+              
+    - name: https
+      required: false
+      description: Use https to connect to prometheus
       value:
-        type: integer
-        example: 7474
-        display_default: 7474
-    - name: server_name
+              type: boolean
+              example: false
+
+    - name: neo4j_version
       required: true
-      description: |
-        Name of the Neo4j server to connect to.
+      description: | 
+                   The Neo4j version to gather metric from. 
+                   Supported versions are 3.5, 4.0, 4.1, 4.2, 4.3
+      value:
+              type: string
+              example: "4.3"
+
+    - name: neo4j_dbs
+      required: false
+      description: | 
+                   The Neo4j databases to gather metric from. 
+                   This is a Neo4j 4.x feature and it has no effect if neo4j_version is set to 3.5.
+      value:
+              type: string
+              example: <DB_1>
+
+    - name: exclude_labels
+      required: false
+      description: | 
+                   A list of prometheus labels to exclude. Those labels will just not be added 
+                   as tags when submitting the metric.
+      value:
+              type: string
+              example: <TAG_1>
+
+    - name: tags
+      required: false
+      description: | 
+                   A list of tags to attach to every metric and service check emitted by this instance. 
+                   
+                   Learn more about tagging at https://docs.datadoghq.com/getting_started/agent/
+      value:
+              type: string
+              example: <KEY_1>:<VALUE_1>
+
+    - name: min_collection_interval
+      required: false
+      description: This changes the collection interval of the check.
+      value:
+              type: integer
+              example: 15
+
+    - name: empty_default_hostname
+      required: false
+      description: This forces the check to send metrics with no hostname.
```

You can utilize configuration templates to provide users will all the configuration options available.

```suggestion
  - template: instance/openmetrics_legacy
```

Then run `ddev validate config -s`.